### ISSUE TITLE
Add per-guild avatar support

### ISSUE THREAD

**Issue by JustAnyones** [Rapptz/discord.py#7054]
### Summary

Add support for per-guild member avatars

### What is the feature request for?

The core library

### The Problem

See https://github.com/discord/discord-api-docs/pull/3081.
Feature implementation would allow bots to access per-guild member avatars.

### The Ideal Solution

Ideally, it would be implemented as a member object method like ``Member.avatar_url_as()``

### The Current Solution

Right now, I use a very hacky solution which includes _hackily_ getting the full member object and then checking whether the avatar field is available and returning the correct URL based on that.

```py
format = 'png'
size = 256

user_data = await client._get_state().http.get_member(guild.id, member.id)
avatar = user_data["avatar"]
if avatar is None:
    return None
# Basically, replicates Asset._from_avatar behaviour
format = 'gif' if avatar.startswith('a_') else format
return f"https://cdn.discordapp.com/guilds/{guild.id}/users/{member.id}/avatars/{avatar}.{format}?size={size}"
```

**Comment by Rapptz** [comment]
This is being worked on already. Note that v1.x won't get any new features and asset was redesigned in v2 so your proposed API won't make it in.
